packages/addresses/src/index.ts

```diff
@@ -0,0 +1,90 @@
+import { config as mainnetConfig } from '@oasisdex/dma-deployments/configs/mainnet.conf'
+import { config as optimismConfig } from '@oasisdex/dma-deployments/configs/optimism.conf'
+import { config as goerliConfig } from '@oasisdex/dma-deployments/configs/goerli.conf'
+
+import { DeploymentConfig, SystemKeys } from '@oasisdex/dma-deployments/types/deployment-config'
+import { Network } from '@oasisdex/dma-common/utils/network'
+
+type DeployedNetworks = Network.MAINNET | Network.OPT_MAINNET | Network.GOERLI
+// TODO: Make address entries type safe after https://github.com/OasisDEX/oasis-earn-sc/pull/249/files is merged
+export type Addresses = Record<DeployedNetworks, Record<SystemKeys, any>>
+
+export const ADDRESSES: Addresses = {
+  [Network.MAINNET]: {
+    mpa: {
+      core: {
+        ...extractAddressesFromConfig(mainnetConfig.mpa.core),
+      },
+      actions: {
+        ...extractAddressesFromConfig(mainnetConfig.mpa.actions),
+      },
+    },
+    common: {
+      ...extractAddressesFromConfig(mainnetConfig.common),
+    },
+    aave: {
+      v2: {
+        ...(mainnetConfig.aave.v2 ? extractAddressesFromConfig(mainnetConfig.aave.v2) : {}),
+      },
+      v3: {
+        ...extractAddressesFromConfig(mainnetConfig.aave.v3),
+      },
+    },
+    maker: {
+      ...extractAddressesFromConfig(mainnetConfig.maker),
+    },
+  },
+  [Network.OPT_MAINNET]: {
```

Comment:
Can we have a full name `OPTIMISM_MAINNET`? OPT sounds like 'optional'.

Reply:
It's been updated in another PR -> https://github.com/OasisDEX/oasis-earn-sc/pull/249